scripts/tests/test_generate_sitemap.py

```diff
@@ -0,0 +1,113 @@
+import pytest
+
+import shutil, tempfile, xml, os
+from urlparse import urljoin
+from website import settings
+from scripts.generate_sitemap import main
+from osf_tests.factories import AuthUserFactory, PreprintProviderFactory, PreprintFactory, ProjectFactory, \
+    RegistrationFactory, CollectionFactory
+
+@pytest.mark.django_db
+class TestGenerateSitemap:
+
+    def test_all_links_are_included(self):
+        # Create temporary directory for the sitemaps to be generated
+        temp_dir = tempfile.mkdtemp()
+
+        # Set static folder to the path of temp_dir, so that the generator would store the sitemaps under
+        # <path_to_temp_dir>/sitemaps/
+        settings.STATIC_FOLDER = temp_dir
+
+        # Test setup
+        user_1 = AuthUserFactory()
+        user_2 = AuthUserFactory()
+        project_1 = ProjectFactory(creator=user_1)
```

Comment:
It looks like these implicitly become public when you create a preprint off of them. Better to be explicit.